client/packages/common/src/intl/context/browserLanguageDetector.ts

```diff
@@ -0,0 +1,37 @@
+import { CustomDetector } from 'i18next-browser-languagedetector';
+
+export const browserLanguageDetector: CustomDetector = {
+  name: 'omsBrowserLanguageDetector',
+  // the language is now supplied by the user profile
+  // and set on login, so the below code is not actually needed
+  // Implementing the language detector allows the language
+  // to be cached in the browser and retained on page refresh
+  lookup: () => {
+    const found: string[] = [];
+    const add = (languageOrLocale?: string) => {
+      if (!languageOrLocale) return;
+      if (/^[a-z]{2}$/i.test(languageOrLocale)) {
+        found.push(languageOrLocale);
+      } else {
+        const language = languageOrLocale.split('-')[0];
+        if (language) found.push(language);
```

Comment:
you can put this into the first regex? `/^([a-z]{2})(-(.*))?$/`

https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCm14CcYF4YHoAUA9A2gIYC0AXgLoDeATAL4CU2x2AdAFT30D8mA3AFD9QkEABt4LUSADm2BEmQt4AD3jBsAInhgNnAcIhiJU2fISKVazduJhSu+kA